### ISSUE TITLE
External SMTP timeout 504

### ISSUE THREAD

**Issue by Lineflyer** [osTicket/osTicket#6310]
### Prerequisites

* [ ] Can you reproduce the problem in a fresh installation of the "develop" branch?
Not tested
* [x] Do you have any errors in the PHP error log, or javascript console?
None
* [x] Did you check the [osTicket forums](https://forum.osticket.com/)?
Nothing matching found (only 5 years old posts)
* [x] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?
Nothing matching found (only 5 years old posts)

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description

After upgrading my osTicket installation from v1.16.3 to v1.17 I found, that after upgrading the mail IMAP and SMTP have been disabled in settings.
Enabling IMAP went without a problem, however enabling SMTP runs into a timeout (HTTP Error 504) when trying to save the settings. I am using a custom SMTP server on port 465 with credentials "Same as Remote Mailbox", which was working flawless before the upgrade.

I tried with some other mail setups and found, that this is only happening with a certain SMTP server, while others are working.

The "problem" could be, that the mail server is using STARTTLS on port 465 instead of SSL. While this seems to work in v1.16.3 I assume v1.17 expects SSL as soon as 465 is the used port.

I already tried to prefix the hostname with `tls://' as mentioned in some other old post. But this makes no difference.

### Steps to Reproduce

1. Go to Admin Panel
2. Go To E-Mail
3. Access the mailbox setting you want to enable
4. Select the tab "Outgoing (SMTP)
5. Set it to active and try to save

![image](https://user-images.githubusercontent.com/949669/195103744-6b48910a-91cd-413a-a51a-39799b5c5c10.png)

**Expected behavior:** [What you expected to happen]
Normal operation

**Actual behavior:** [What actually happened]
Shows "Loading" toast and runs into gateway timeout (error 504) after some 20-30 seconds
![grafik](https://user-images.githubusercontent.com/949669/195111924-25e24fb4-5964-4488-b89f-5e4217310659.png)

### Versions

Admin panel -> Dashboard -> Information which also additionally gives you information about your server.
![image](https://user-images.githubusercontent.com/949669/195096147-982f1938-c2e0-4b7b-ad91-07278764375e.png)

**Comment by mixneko** [comment]
I also have some problem
but my port is 587
and I can't send to any google mail..........
and use php function to send the mail wall be error

![圖片](https://user-images.githubusercontent.com/12106753/195112663-fab3b87c-eb20-4f84-bc1c-b0e597b2309d.png)

**Comment by Lineflyer** [comment]
> and use php function to send the mail wall be error

Using PHP as mailer is something totally different to using SMTP. My problem is about SMTP.
Unfortunately I am not sure how to get any logging or more info on my problem (php log and system debug log is empty).

**Comment by mixneko** [comment]
I know what you mean
but I'm also have this problem and php function problem

**Comment by Lineflyer** [comment]
FYI:

I performed a fallback (from my webspace + DB backup) to osTicket v1.16.3.
Now SMTP is functional again, which makes it quite obvious, that a change in v1.17 causes the above described problem.

Update to the issue itself (will also adapt original text):
In contrast to what is described above, there are no self signed certificates involved.
The "problem" could be, that the mail server is using STARTTLS on port 465 instead of SSL. While this seems to work in v1.16.3 I assume v1.17 expects SSL as soon as 465 is the used port.

**Comment by JediKev** [comment]
@Lineflyer 

You can see from the below snippet that this should not be the case. `465` is considered SSL and should be using SSL as such:
- https://github.com/osTicket/osTicket/blob/1.17.x/include/class.mail.php#L816-L817

Cheers.

**Comment by Lineflyer** [comment]
@JediKev 

Do I read the code correct:
If port is 465 it will always use SSL despite prefixing tls:// to the hostname?

**Comment by ChrisTG742** [comment]
[Here's a SO article](https://stackoverflow.com/questions/15796530/what-is-the-difference-between-ports-465-and-587) about the difference between port 465 and port 587.

TL;DR: The only ports that support un-encrypted traffic are the old SMTP-port 25 and the "submission" port 587. The latter additionally supports starting TLS via STARTTLS. Port 465 **directly** starts encrypted, so the TLS-handshare occurs right after the connect. Just as it is with HTTPS.

**Comment by Lineflyer** [comment]
Small update on this one:
I did another attempt of upgrading osticket from v1.16.3 to v1.17.2.

The problem described above still exists:
- Before the upgrade I was succesfully using SMTP with SSL on port 465
- After the upgrade (and without changing mail configuration) it ceased to work. No mails outgoing.
- Only after falling back to port 25 it works

Any other insights or any clue how to debug this on osticket side? There is no proper error message...its just running into the 504 error.
